hibernate-core/src/main/java/org/hibernate/persister/collection/BasicCollectionPersister.java

```diff
@@ -597,19 +597,19 @@ private RestrictedTableMutation<JdbcMutationOperation> generateDeleteRowAst(Muta
 		final TableDeleteBuilderStandard deleteBuilder = new TableDeleteBuilderStandard( this, tableReference, getFactory() );
 
 		if ( pluralAttribute.getIdentifierDescriptor() != null ) {
-			deleteBuilder.addKeyRestriction( pluralAttribute.getIdentifierDescriptor() );
+			deleteBuilder.addKeyRestrictionLeniently( pluralAttribute.getIdentifierDescriptor() );
 		}
 		else {
 			pluralAttribute
 					.getKeyDescriptor()
 					.getKeyPart()
-					.forEachSelectable( (index, selectable) -> deleteBuilder.addKeyRestriction( selectable ) );
+					.forEachSelectable( (index, selectable) -> deleteBuilder.addKeyRestrictionLeniently( selectable ) );
```

Comment:
A general observation as this happens a bunch - 

We should add a form of `#addKeyRestrictionLeniently` which accepts the index as well.  Even though it will ignore the index, that will allow the use of a method reference as opposed to lambdas

Reply:
it was already there but not used in this case :( fixed this
I added instead one for `addKeyRestriction`